clicache/src/ReflectionBasedAutoSerializer.cpp

```diff
@@ -536,20 +536,14 @@ namespace Apache
           {
             return FieldType::ARRAY_OF_BYTE_ARRAYS;
           }
-          /*else if(type->Equals(Internal::DotNetTypes::ObjectArrayType))
-          {
-            //Giving more preference to arraylist instead of Object[] in java side
-            //return this->WriteObjectArray(fieldName, safe_cast<System::Collections::Generic::List<Object^>^>(fieldValue));
-            return FieldType::OBJECT_ARRAY;
-          }*/
-          else
+          else if(type->Equals(Internal::DotNetTypes::ObjectType))
```

Comment:
Does this check fail for primitive types then? What about the object version of the primitive types, won’t that pass this test and still cause issues?

Reply:
There are no breaking changes in this PR. Support for new types have been implemented in new APIs.